lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -586,6 +588,32 @@ def install_a_puppet_msi_on(hosts, opts)
           end
         end
 
+        # Installs Puppet and dependencies from FreeBSD ports
+        #
+        # @param [Host, Array<Host>, String, Symbol] hosts    One or more hosts to act upon,
+        #                            or a role (String or Symbol) that identifies one or more hosts.
+        # @param [Hash{Symbol=>String}] opts An options hash
+        # @option opts [String] :version The version of Puppet to install (shows warning)
+        #
+        # @return nil
+        # @api private
+        def install_puppet_from_freebsd_ports_on( hosts, opts )
+          if (opts[:version])
+            logger.warn "FreeBSD's port system doesnt accept versions, using latest"
```

Comment:
Yeah, it means that you can have code that asks for a specific version but it wont bomb out if it's on a FreeBSD hosts